src/codebase.md

```diff
@@ -260,3 +260,41 @@ crates optimally for your semantic needs.
 | [ring](https://docs.rs/ring/)           | the leading crypto library         |
 | [nalgebra](https://docs.rs/nalgebra/)   | linear algebra                     |
 | [once_cell](https://docs.rs/once_cell/) | complex static data                |
+
+## How should I call C++ functions from Rust and vice-versa?
+
+Behind the scenes, Rust natively supports calling C functions using `extern "C"`,
+`#[repr(C)]` and `#[no_mangle]`. This is, of course, error prone and involves
+lots of `unsafe` code. So don't do that: there are better ways.
+
+Your initial decision should be a matter of policy. Choose between:
+
+* I want to manually specify the cross-language interface in order to keep
+  a degree of isolation and to avoid proliferation of uncontrolled
+  cross-language boundaries. Or:
+* I want to generate comprehensive bindings automatically from existing C++
+  headers to allow arbitrary calls from Rust to C++.
+
+If you choose the former approach, the modern standard for C++ interoperability
+is [cxx](https://cxx.rs). You specify cross-language interfaces in a
+Rust-like interface definition language (within your Rust file). Both Rust and
```

Comment:
Add "(IDL)" after "interface definition language" to introduce this abbreviation.

Reply:
Done